docs/docs/contribute/document-minimal-permissions.mdx

```diff
@@ -0,0 +1,223 @@
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Documenting minimal permissions
+
+To help you improve your security posture, we're working on documenting minimal permissions required for a command to run. Using this information, you can then create a new Entra app registration with only the permissions required for your script to run. Documenting the minimal permissions is a work in progress and we'd appreciate your help. This page describes how to detect the minimal permissions for a CLI for Microsoft 365 command, and update the permissions reference in case some APIs are missing.
+
+## Detect minimal permissions
+
+To detect the minimal permissions for a command, we use [Dev Proxy](https://aka.ms/devproxy). Dev Proxy records the API requests that a command makes. Then, it analyzes the requests to determine the minimal permissions required for the command to run. For the analysis, it uses information about Microsoft Graph permissions from its documentation. For other APIs, such as SharePoint REST and CSOM, it uses information that we [curate together with the community](https://adoption.microsoft.com/sample-solution-gallery/sample/pnp-devproxy-sharepoint-api-minimal-permissions).
+
+To detect the minimal permissions for a command:
+
+### Start Dev Proxy
+
+<Tabs groupId="shell">
+  <TabItem label="PowerShell" value="powershell">
+    - Open a terminal.
+    - Change the working directory to the CLI for Microsoft 365 repo root.
+    - Start Dev Proxy with the permission detection preset:
+
+      ```PowerShell
+      npm run devproxy:permissions -- -- -e "tenant=<your_tenant>"
+      ```
+      
+      For example, if your tenant is `contoso.onmicrosoft.com` run:
+      ```PowerShell
+      npm run devproxy:permissions -- -- -e "tenant=contoso"
+      ```
+
+  </TabItem>
+  <TabItem label="zsh/bash" value="sh">
+    - Open a terminal.
+    - Change the working directory to the CLI for Microsoft 365 repo root.
+    - Start Dev Proxy with the permission detection preset:
+      ```bash
+      npm run devproxy:permissions -- -e="tenant=<your_tenant>"
+      ```
+      
+      For example, if your tenant is `contoso.onmicrosoft.com` run:
+      
+      ```bash
+      npm run devproxy:permissions -- -e="tenant=contoso"
+      ```
+
+  </TabItem>
+</Tabs>
+
+### Run the CLI command through proxy
+
+<Tabs groupId="shell">
+  <TabItem label="PowerShell" value="powershell">
+    - Open a new terminal.
+    - Set environment variables:
+
+      ```PowerShell
+      $env:HTTPS_PROXY="http://127.0.0.1:8000"
+      $env:NODE_TLS_REJECT_UNAUTHORIZED=0
+      ```
+
+    - Run the command:
+
+      ```PowerShell
+      m365 spo web list -u /
```

Comment:
Using `m365 spo web list -u "https://contoso.sharepoint.com"` would be a better example, since the trailing `/` only works if the user has previously set their SPO URL using `m365 spo set`.

Reply:
Good idea